packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -26,39 +23,40 @@ export class RootDataObjectWithChildDataObject extends DataObjectWithCounter {
     protected async hasInitialized(): Promise<void> {
         await super.hasInitialized();
         await this.createAndReferenceChild();
-        // TODO: Start logic goes here after we agree on what it should be.
+        this.start();
     }
 
-    // TODO: Make private - left public so the build passes
-    public async unreferenceChild() {
+    private unreferenceChild() {
         assert(this.child !== undefined, `Child should be defined when unreferencing!`);
-        this.unreferencedChild = this.child;
         this.child.stop();
         this.root.delete(this.childKey);
-        this.unrefTimeStampMs = Date.now();
     }
 
     private async createAndReferenceChild() {
         this.child = await dataObjectWithCounterFactory.createInstance(this.context.containerRuntime);
-        await this.referenceChild(this.child.handle);
+        this.root.set(this.childKey, this.child.handle);
+        this.child.start();
     }
 
-    private async referenceChild(handle: IFluidHandle<DataObjectWithCounter>) {
-        this.unreferencedChild = this.child;
-        this.root.set(this.childKey, handle);
-    }
+    protected async sendOps() {
```

Comment:
I feel like there could be a way to pull the loop logic down to the base class and let subclasses just define the step to take each cycle.  It's basically there, if you keep `sendOps` private and move all this logic to `sendOp`.

All that's missing is the base class providing a "local op" count.  Or if it works for your logic, you could just use the counter itself - inside `sendOp` in this class, do:

```typescript
if (this.counter.value % this.opsToWait === 0) {
            // Reference the child if there is none, unreference if there is one
            if (this.child === undefined) {
                await this.createAndReferenceChild();
            } else {
                this.unreferenceChild();
            }
}
```

Or something? Feel free to punt on it, just wanted to bring it up.

Reply:
I'll punt on this, this would be another layer of thinking, this would get weird with multiple clients on. The more clients there are, the faster we are referencing and unreferencing.